pdns/dnsdist-lua-rules.cc

```diff
@@ -738,6 +739,52 @@ class RCodeRule : public DNSRule
   int d_rcode;
 };
 
+class ERCodeRule : public DNSRule
+{
+public:
+  ERCodeRule(int rcode) : d_rcode(rcode & 0xF), d_extrcode(rcode >> 4)
+  {
+  }
+  bool matches(const DNSQuestion* dq) const override
+  {
+    // avoid parsing EDNS OPT RR when not needed.
+    if (d_rcode != dq->dh->rcode) {
+      return false;
+    }
+
+    char * optStart = NULL;
+    size_t optLen = 0;
+    bool last = false;
+    int res = locateEDNSOptRR((char*)dq->dh, dq->len, &optStart, &optLen, &last);
```

Comment:
`const_cast<char*>(reinterpret_cast<const char*>(dq->dh))` ? I don't think we'll manage to get rid of the existing C-style casts, but it would be nice to add new ones as few as possible :)